standard/expressions.md

```diff
@@ -4822,7 +4822,7 @@ The null coalescing operator is right-associative, meaning that operations are g
 
 The type of the expression `a ?? b` depends on which implicit conversions are available on the operands. In order of preference, the type of `a ?? b` is `A₀`, `A`, or `B`, where `A` is the type of `a` (provided that `a` has a type), `B` is the type of `b`(provided that `b` has a type), and `A₀` is the underlying type of `A` if `A` is a nullable value type, or `A` otherwise. Specifically, `a ?? b` is processed as follows:
 
-- If `A` exists and is not a nullable value type or a reference type, a compile-time error occurs.
+- If `A` exists and is not a nullable value type, a compile-time error occurs.
```

Comment:
I’ll review this properly later but for now did you mean:

```suggestion
- If `A` exists and is a non-nullable value type a compile-time error occurs.
```

as the current wording would *exclude* reference types? Should other types be invalid, e.g. pointer types?

Reply:
I share the same confusion as Nigel here. And I don't know offhand whether a type parameter that is constrained (e.g. `T where T : struct`) counts as "a non-nullable value type" or just "known to be a non-nullable value type".

Reply:
I applied Nigel's suggestion.

I unresolved the conversation because of Jon's question.

Reply:
I concur Bill; if I recall correctly the intention of the C# 2.0 design was that `where T:struct` definitely ensures that `T` is a non-nullable value type.